src/depotchest.cpp

```diff
@@ -27,8 +27,8 @@ ReturnValue DepotChest::queryAdd(int32_t index, const Thing& thing, uint32_t cou
 			addCount = 1;
 		}
 
-		if (item->getTopParent() != this) {
-			if (const Container* container = item->getContainer()) {
+		if (item->getTopParent().get() != this) {
```

Comment:
Could this be `const auto`
```suggestion
		if (item->getTopParent() != getContainer()) {
```